src/ResourceManager/Resources/Commands.Resources/Models.ActiveDirectory/ActiveDirectoryClient.cs

```diff
@@ -251,8 +251,17 @@ public List<PSADObject> ListUserGroups(string principal)
         public List<PSADObject> GetObjectsByObjectId(List<string> objectIds)
         {
             List<PSADObject> result = new List<PSADObject>();
-            var adObjects = GraphClient.Objects.GetObjectsByObjectIds(new GetObjectsParameters { ObjectIds = objectIds, IncludeDirectoryObjectReferences = true });
-            result.AddRange(adObjects.Select(o => o.ToPSADObject()));
+            IPage<AADObject> adObjects;
+            for(int i=0; i<objectIds.Count; i+=1000)
+            {
+                if((i+1000) > objectIds.Count){
+                    adObjects = GraphClient.Objects.GetObjectsByObjectIds(new GetObjectsParameters { ObjectIds = objectIds.GetRange(i, objectIds.Count - i), IncludeDirectoryObjectReferences = true });
```

Comment:
>objectIds.Count - i [](start = 135, length = 19)

compute the end index in "if-else" block then merge two graph calls into one line